src/compiler/emitter.ts

```diff
@@ -2302,35 +2420,52 @@ namespace ts {
         function emitModifiers(node: Node, modifiers: NodeArray<Modifier>) {
             if (modifiers && modifiers.length) {
                 emitList(node, modifiers, ListFormat.Modifiers);
-                write(" ");
+                writeSpace();
+            }
+        }
+
+        function emitAsTypeAnnotation(node: Node) {
+            if (node) {
+                writePunctuation(":");
+                writeSpace();
+                emit(node);
+            }
+        }
+
+        function emitAsAssignmentTarget(node: Expression) {
+            if (node) {
+                writeSpace();
+                writeOperator("=");
+                writeSpace();
+                emitExpression(node);
             }
         }
 
-        function emitWithPrefix(prefix: string, node: Node) {
-            emitNodeWithPrefix(prefix, node, emit);
+        function emitWithPrefix(prefix: string, writer: (s: string) => void, node: Node) {
+            emitNodeWithPrefix(prefix, writer, node, emit);
         }
 
-        function emitExpressionWithPrefix(prefix: string, node: Node) {
-            emitNodeWithPrefix(prefix, node, emitExpression);
+        function emitExpressionWithPrefix(prefix: string, writer: (s: string) => void, node: Node) {
+            emitNodeWithPrefix(prefix, writer, node, emitExpression);
         }
 
-        function emitNodeWithPrefix(prefix: string, node: Node, emit: (node: Node) => void) {
+        function emitNodeWithPrefix(prefix: string, writer: (s: string) => void, node: Node, emit: (node: Node) => void) {

```

Comment:
Could we add a default initializer for `emit` to eliminate the similarly-named `emitWithPrefix`?

Reply:
Sure. `emitWithPrefix` is only used in 3 places anyway.